### PR TITLE
Add a script to update the stepMode of all maze levels that only show…

### PR DESCRIPTION
**Pull request by trevorb**
Kiki has asked that we show the step button on all maze levels.  This script goes through and updates all maze levels that only show the run button to also show the step button.

### CODE REVIEW

File: dashboard/scripts/maze_show_step_button.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env ruby
+
+# This script updates all maze levels that are configured to
+# only show the run button to also show the step button
+
+require_relative '../config/environment'
+
+puts "Starting"
+
+mazes = Level.all.select { |x| x.is_a? Maze }
```
Comment (laurelfan) [inline dashboard/scripts/maze_show_step_button.rb]: does `Maze.all` not work?

**Comment by laurelfan** [comment]
For things like this with manual deployment steps we have been starting to use the github checklist feature (see https://github.com/blog/1375%0A-task-lists-in-gfm-issues-pulls-comments if you're not familiar with this), so you could do something like:

- [ ] run this on levelbuilder
- [ ] check in modified .level files

**Comment by trevorb** [comment]
Cool, I didn't realize you could do Maze.all

I read the blog post you linked to but I guess I'm still not clear how I should use the task list for something like this. I thought after this gets pushed I would run it on a production machine.  Can you explain a bit more?

**Comment by laurelfan** [comment]
Ah... Let me explain the level builder process to you tomorrow (or ask Brian/Will if that's most convenient). It's a bit complicated, but the short version: the "real" level data is in .level files that are imported on every deploy. If you only change the production database your change will be overwritten. The levelbuilder instance is a "private" instance running the same code with slightly different settings where the ed team does their level editing -- this instance is capable of writing .level files so it may make the most sense for you to run this script there and then check in the changed files so they are imported on the next production deploy.

**Comment by trevorb** [comment]
Oh, I see I didn't realize the .level files were the "master" copy. So the process would be 

- [ ] Notify the levelbuilders room that I'm going to be updating levelbuilder levels
- [ ] Run the script on level builder machine
- [ ] Check in the changes to the level files
- [ ] Push the changes back to github

I think thats it?

**Comment by laurelfan** [comment]
yes, that looks good. If Kiki has a list of levels (or a list of stages) that she does not want this change applied to, maybe your script can actually handle checking for those and not changing them?

**Comment by trevorb** [comment]
good idea, I'll ask kiki

On Tue, Sep 15, 2015 at 2:36 PM, Laurel <notifications@github.com> wrote:

> yes, that looks good. If Kiki has a list of levels (or a list of stages)
> that she does not want this change applied to, maybe your script can
> actually handle checking for those and not changing them?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/code-dot-org/code-dot-org/pull/3978#issuecomment-140554036>
> .
>



-- 
Trevor
trevor@code.org

**Pull request merged**